### PR TITLE
VM & VMSS MSI, GetImage with 'latest' version & Moving new methods to Beta

### PR DESCRIPTION
**Pull request by anuchandy**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by anuchandy** [comment]
@jianghaolu could you review this [commit](https://github.com/Azure/azure-sdk-for-net/commit/fe337961db1873c3e9f3da3c613a3e45cffc436f)? This was required to make RBAC role assignment work for service principal assigned to VM.

File: Samples/Samples.csproj [hunk 1]
```diff
@@ -56,7 +56,6 @@
     <Folder Include="AppService\" />
     <Folder Include="Batch\" />
     <Folder Include="Cdn\" />
-    <Folder Include="Compute\" />
```
Comment (hovsepm) [inline Samples/Samples.csproj]: ><Folder Include="Compute\" /> [](start = 4, length = 29)

Why is this file changed?

File: src/ResourceManagement/Compute/VirtualMachineScaleSetMsiHelper.cs [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+                {
+                    result = (int)obj;
+                }
+                else if (obj is Int64)
+                {
+                    result = (int)((Int64)obj);
+                }
+                else
+                {
+                    result = int.Parse((string)obj);
```
Comment (hovsepm) [inline src/ResourceManagement/Compute/VirtualMachineScaleSetMsiHelper.cs]: >Parse [](start = 33, length = 5)

this will throw if the object is invalid int. there should be a method TryParse that should match your use case.

File: src/ResourceManagement/Compute/VirtualMachineMsiHelper.cs [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+        }
+
+        /// <summary>
+        /// Given an object holding a numeric in Integer or String format, convert that to
+        /// Integer.
+        /// </summary>
+        /// <param name="obj">The object.</param>
+        /// <return>The integer value.</return>
+        ///GENMHASH:663112EB0F1DE517DFEE2A837DECC2B6:E55F34EE6CDFC6C04AA107D22EE30395
+        private static int? ObjectToInteger(object obj)
```
Comment (hovsepm) [inline src/ResourceManagement/Compute/VirtualMachineMsiHelper.cs]: > private static int? ObjectToInteger(object obj) [](start = 7, length = 48)

you already has this method in VirtualMachineScalreSetsImpl, right?

**Review by hovsepm — CHANGES_REQUESTED** [review]
Please address review comments

**Review by hovsepm — APPROVED** [review]

**Review by jianghaolu — APPROVED** [review]
